src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/params.py

```diff
@@ -76,3 +76,25 @@
     c.register_alias('name', ('--name', '-n'))
     c.ignore('user_name')
     c.argument('lab_name', validator=validate_user_name)
+
+with ParametersContext(command='lab secret create') as c:
+    from .sdk.devtestlabs.models.secret import Secret
+    c.expand('secret', Secret)
+
+    c.ignore('id')
+    c.ignore('location')
+    c.ignore('type')
+    c.ignore('provisioning_state')
+    c.ignore('unique_identifier')
+    c.ignore('tags')
+
+with ParametersContext(command='lab secret update') as c:
+    from .sdk.devtestlabs.models.secret import Secret
+    c.expand('secret', Secret)
```

Comment:
Yes, you shouldn't use expand here. Use a validator to convert the parameter string to a Secret object (exactly like your custom command does)